packages/x-date-pickers-pro/src/DateRangePickerDay/DateRangePickerDay.tsx

```diff
@@ -136,12 +145,10 @@ const DateRangePickerDayRoot = styled('div', {
   ...((ownerState.isStartOfHighlighting || ownerState.isStartOfMonth) && {
     ...startBorderStyle,
     paddingLeft: 0,
-    marginLeft: DAY_MARGIN / 2,
```

Comment:
I've removed these styles to fix the following inconsistency (currently available, but less noticeable while dragging is not supported) when the highlighting range changes.
Do you @alexfauquette @flaviendelangle  see any potential issues with it? 🤔 

https://user-images.githubusercontent.com/4941090/203561310-62940c7d-c30e-4766-8622-0b1028991017.mov


https://user-images.githubusercontent.com/4941090/203561336-0480cc31-4b59-469c-8722-29ff2d51d910.mov

Reply:
Did you not see the weekdays jumping in width in the first video when we are dragging in them? 🤔

Reply:
Amazing, I don't know how I made to miss it 👀

Reply:
It's great to fix these jumping
Did you see any down side of removing this margin ?

Reply:
I'll do one more testing round with different variations (mobile, 1 and several calendars). 😉

Reply:
Seems like this is a fix we needed anyways.
All instances of calendars seems to be missing this alignment adjustment.

**Mobile before:**
![mobile-before](https://user-images.githubusercontent.com/4941090/203802935-cea63069-5e6c-4eb8-bc6e-8066f1e4b261.png)
**Mobile after:**
![mobile-after](https://user-images.githubusercontent.com/4941090/203802971-f2966d09-4c4f-469d-bdb2-7aaf125b1467.png)
**Desktop before:**
![desktop-before](https://user-images.githubusercontent.com/4941090/203802985-6b0e0e69-2358-49c7-a2fb-5886582dd439.png)
**Mobile after:**
![desktop-after](https://user-images.githubusercontent.com/4941090/203802998-619e6f30-aa8b-45d5-a0d7-1eb693c28c9b.png)